e2e/tasks_test.go

```diff
@@ -0,0 +1,146 @@
+// +build e2e
+
+package e2e
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul-terraform-sync/testutils"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTasksUpdate(t *testing.T) {
+	// Tests multiple tasks are triggered on service registration and de-registration
+	// by verifying the content of terraform.tfvars
+	t.Parallel()
+
+	srv := newTestConsulServer(t)
+	defer srv.Stop()
+
+	tempDir := fmt.Sprintf("%s%s", tempDirPrefix, "multiple_tasks")
+	delete := testutils.MakeTempDir(t, tempDir)
+	testutils.MakeTempDir(t, tempDir)
+
+	apiTaskName := "e2e_task_api"
+	apiTask := fmt.Sprintf(`
+task {
+	name = "%s"
+	description = "basic read-write e2e task api only"
+	services = ["api"]
+	providers = ["local"]
+	source = "../../test_modules/e2e_basic_task"
+}
+`, apiTaskName)
+	configPath := filepath.Join(tempDir, configFile)
+	var config hclConfig
+	config = config.appendConsulBlock(srv).appendTerraformBlock(tempDir).
+		appendDBTask().appendWebTask().appendString(apiTask)
```

Comment:
ah! love this use of `appendString()` to add one-off custom configs! I will have to steal this :)